shared/modules/Storage/ReadOnlyNetworkStore.ts

```diff
@@ -0,0 +1,115 @@
+import log from 'loglevel';
+import { isErrorWithMessage } from '@metamask/utils';
+import getFetchWithTimeout from '../fetch-with-timeout';
+import type Migrator from '../../../app/scripts/lib/migrator';
+import { type IntermediaryStateType, BaseStorage } from './Storage';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_URL = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}/state.json`;
+
+/**
+ * A read-only network-based storage wrapper
+ */
+export default class ReadOnlyNetworkStore extends BaseStorage {
+  #initialized: boolean;
+
+  #promiseToInitialize?: Promise<void>;
+
+  #state: IntermediaryStateType | null;
+
+  mostRecentRetrievedState: IntermediaryStateType | null;
+
+  stateCorruptionDetected: boolean;
+
+  dataPersistenceFailing: boolean;
+
+  migrator: Migrator;
+
+  firstTimeInstall: boolean;
+
+  constructor({ migrator }: { migrator: Migrator }) {
+    super();
+    this.#initialized = false;
+    this.#promiseToInitialize = this.#init();
+    this.#state = null;
+    this.mostRecentRetrievedState = null;
+    this.stateCorruptionDetected = false;
+    this.dataPersistenceFailing = false;
+    this.migrator = migrator;
+    this.firstTimeInstall = false;
+  }
+
+  /**
+   * Declares this store as compatible with the current browser
+   */
+  isSupported = true;
+
+  /**
+   * Initializes by loading state from the network
+   */
+  async #init() {
+    try {
+      const response = await fetchWithTimeout(FIXTURE_SERVER_URL);
+      if (response.ok) {
+        this.#state = await response.json();
+      }
+    } catch (error) {
+      if (isErrorWithMessage(error)) {
```

Comment:
Could we ever have an error without a message here? If so, why would we ignore it?

Reply:
Addressed in ada46a448b